Instrukcije_d001/src/app/terminal.c

```diff
@@ -0,0 +1,131 @@
+//
+// Created by Edin on 15. 03. 2022.
+//
+
+#include "terminal.h"
+
+int stevec = 0; // counter za to kdo bo na vrsti (X ali 0)
+int igreJeKonec = 0; // flag za konec igre
+
+void terminal_init() {
+    initscr();
+    curs_set(0);
+
+    noecho();
+    refresh();
+}
+
+// Zakaj memoTabela mora bit const?
+void jeKdoZmagal(int tabela[ST_ZMAGOVALNIH_KOMBINACIJ][3], int memoTabela[WORLD_ST_ZETONOV]) {
+    for (int i = 0; i < ST_ZMAGOVALNIH_KOMBINACIJ; ++i) {
+        if ((memoTabela[tabela[i][0]] > 0 && memoTabela[tabela[i][1]] > 0 && memoTabela[tabela[i][2]] > 0)
+            && (memoTabela[tabela[i][0]] == memoTabela[tabela[i][1]])
+            && (memoTabela[tabela[i][0]] == memoTabela[tabela[i][2]])) {
+            igreJeKonec = 1;
+            mvprintw(8, 0, "Game over!\nPlayer %c has won the game!\n"
+                           "Press any key to exit.", (stevec % 2 == 0) ? 'O' : 'X');
+        }
+    }
+}
+
+void poteza(World* world, int* memoTabela, int c) {
+    int zaVnest = (stevec % 2) ? 1 : 2;
+    memoTabela[c] = zaVnest;
+
+    char XorO = (stevec % 2) ? 'O' : 'X';
+
+    Zeton zeton = zeton_new(c % 3, c / 3, XorO);
+    world->zetoni[stevec] = zeton; // dodelimo tabeli zetonov zeton
+
+    stevec++;
+}
+
+int terminal_main() {
+    World world = world_new();
+
+    int* memoTabela = (int*) calloc(WORLD_ST_ZETONOV, sizeof(int)); // za prevert ce je kdo zmagal
+    int zmagovalneKombinacije[7][3] = {{0,1,2},{3,4,5},{6,7,8},{0,3,6},{1,4,7},{0,4,8},{2,4,6}};
```

Comment:
Da zapišeš vse kombinacije v eno tabelo je ok če imaš 3*3 tabelo vendar to odpove če imaš poljubno polje n*m kar bo problematično ko bom želel od tebe da narediš poljubno velikost polja. 😄